src/cpp/session/modules/SessionHelp.R

```diff
@@ -92,12 +92,50 @@ options(help_type = "html")
    list(payload, "text/html", character(), 404)
 });
 
-.rs.addJsonRpcHandler("suggest_topics", function(prefix)
+.rs.setVar("topicsEnv", new.env(parent = emptyenv()))
+
+.rs.addJsonRpcHandler("suggest_topics", function(query)
 {
-   if (getRversion() >= "3.0.0")
-      sort(utils:::matchAvailableTopics("", prefix))
-   else
-      sort(utils:::matchAvailableTopics(prefix))
+   pkgpaths <- path.package(quiet = TRUE)
+   
+   # read topics from
+   topics <- lapply(pkgpaths, function(pkgpath) tryCatch({
+      
+      if (exists(pkgpath, envir = .rs.topicsEnv))
+         return(get(pkgpath, envir = .rs.topicsEnv))
+      
+      aliases <- file.path(pkgpath, "help/aliases.rds")
+      index <- file.path(pkgpath, "help/AnIndex")
+      
+      value <- if (file.exists(aliases)) {
+         names(readRDS(aliases))
```

Comment:
If I'm reading this correctly we will be re-parsing the RDS files from disk, for every installed package, every time the user adds letters to their search string. I think that's going to crawl on non-SSDs and we should probably cache here.

Reply:
We do check for a cached value here:

https://github.com/rstudio/rstudio/pull/5666/files#diff-3ac392a1e062b2834094488029b1bb8bR104-R105

And store the computed value in the cache here:

https://github.com/rstudio/rstudio/pull/5666/files#diff-3ac392a1e062b2834094488029b1bb8bR117

Reply:
You're right, of course, sorry I missed that!